library/Reporting/Web/Forms/ReportForm.php

```diff
@@ -124,6 +157,34 @@ protected function assemble()
         }
     }
 
+    public function validate()
+    {
+        parent::validate();
+        if (! $this->isValid) {
+            return false;
+        }
+
+        if ($this->id || $this->getPopulatedValue('remove')) {
+            return true;
+        }
+
+        $name = $this->getValue('name');
```

Comment:
Sure that is nice, but I would be perfectly fine if you remove that completely. I'm not aware that we do this consistently in our products. I think we just rely on the database to complain about duplicate key errors. If you really want to add this, try to avoid overriding `validate` and attach a `CallbackValidator` in `assemble()` if `id` is `null`.